### PR TITLE
LPS-119007 | master

### PR DESCRIPTION
**Pull request by john-co**
### JIRA Ticket:
https://issues.liferay.com/browse/LPS-119007
### Description:
We don't need to rely on the UI to test if something went wrong with frontend token. However, we would like to include a usecase test as part of `ci:test:relevant` coverage just in case a change in frontend token breaks Style books. We'll add one of the Style Books tests that is the most basic: ChangeConfigurationOfPageBasedOnStyleBook.
### Test Plan:
1. Add a StyleBooks#ChangeConfigurationOfPageBasedOnStyleBook into front-end token relevant scope

### CODE REVIEW

**Comment by john-co** [comment]
ci:test:relevant passed in https://github.com/john-co/liferay-portal/pull/388#issuecomment-797840093

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/stylebooks/StyleBooks.testcase [hunk 1]
```diff
@@ -241,6 +241,8 @@ definition {
 	@description = "This is for LPS-116106. Page Creator can change style book of content page via Look and Feel settings."
 	@priority = "4"
 	test ChangeStyleBookOfContentPageViaLookAndFeel {
+		property testray.component.names = "Frontend Token";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/stylebooks/StyleBooks.testcase:244]: This modification will override the original component

**Comment by jbalsas** [comment]
> This includes some style books testcases in ci:test:relevant for changes done to frontend token

Hey @john-co, could you describe a bit more in detail how does this work? Which `testcases` are added. I assume is just `ChangeStyleBookOfContentPageViaLookAndFeel`?

My main concern is that this test does (roughly):
- 🚫  Add a style book
- ⚠️  Edit the style book and Publish it
- 🚫  Add a content page
- 🚫  Add a Heading fragment to page
- ⚠️  Configure the Heading Level of Heading as H5
- 🚫  Add a Button fragment below the Heading
- ⚠️  Configure the Type of Heading as Button Outline Secondary
- 🚫  Change to the new style book via Look and Feel
- 🚫  Assert the new style book is applied to content page in view mode (this depends on **frontend-css-variables**, not frontend-token)
- 🚫  Change to default style book via Look and Feel
- 🚫  Assert the default style book is applied to content page in view mode (this depends on **frontend-css-variables**, not frontend-token)

I've marked as ⚠️  the steps that are marginally connected to tokens and with 🚫  the ones that aren't likely to be related. Based on this, my main concerns:
- StyleBook failures making tokens fail for unrelated reasons
- No tests of our own, no way to assert anything about the quality of our module unless done through direct integration with other apps

**Comment by john-co** [comment]
> No tests of our own, no way to assert anything about the quality of our module unless done through side-effects when integrated in other apps

We already have java unit tests of our own to cover the frontend token API:
https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/4165/AXIS_VARIABLE=0,label_exp=!master/testReport/com.liferay.frontend.token.definition.internal/

> StyleBook failures making tokens fail for unrelated reasons

I think there's some value in having usecase tests in addition to our java unit tests for more comprehensive coverage of our real customer usecases.

If there's a failure due to some other team that fails those usecase tests then CI should normally be able to exclude them from unique test failures. If test is identified as flaky, then imo only 3 tests is manageable to isolate or reach out to echo team/qa to quickly review.

> could you describe a bit more in detail how does this work?

Based on the filter from test.properties file,
it will run this test from the following query `ChangeStyleBookOfContentPageViaLookAndFeel`
```
(testray.component.names == "Frontend Token")
```
and in addition to catch tests that may affect acceptance Style Book tests we include these tests with this filter:
- `ChangeConfigurationOfPageBasedOnStyleBook`
- `PreviewTheEffectOnPageInStyleBookEditor`

```
OR \
        (\
            (portal.acceptance == true) AND \
            (testray.main.component.name ~ "Style Books")\
        )
```

**Comment by jbalsas** [comment]
> I think there's some value in having usecase tests in addition to our java unit tests for more comprehensive coverage of our real customer usecases.

I'm not arguing wether there's value or not. My main concern is wether there's **enough** value to compensate for longer, flakier CI runs and wether the expectations we have for what this test is going to catch are founded.

It should be enough to specify X, Y and Z as potential errors that we've identified based on the connection between tokens and stylebooks that we expect this additional test to protect us from. 

> and in addition to catch tests that may affect acceptance Style Book tests we include these tests with this filter:
> - ChangeConfigurationOfPageBasedOnStyleBook
> - PreviewTheEffectOnPageInStyleBookEditor
I think the first one might be related to tokens but the second one most likely not.

**Comment by john-co** [comment]
I see ok, I'll spend some time to analyze. I do want to have a better balance of our test steps to be more related to the frontend-token module so maybe just having ChangeConfigurationOfPageBasedOnStyleBook as a usecase test would be enough to cover my concern of no usecase test with frontend-token. We'll also try to document this concern on our ticket/github comment better.

cc/ @calvinctvo @zhangyan0701 please take note of this too.

**Comment by john-co** [comment]
ci:test:relevant

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**